include/materials/modified_cam_clay.tcc

```diff
@@ -540,7 +503,8 @@ void mpm::ModifiedCamClay<Tdim>::compute_df_dsigma(
     // Compute r
     double r_val = 0.;
     // Compute j2
-    double j2 = 3 * std::pow(q, 2);
+    const double j2 = mpm::materials::j2(stress);
+    const double j3 = mpm::materials::j3(stress);
```

Comment:
I don't think you need this anymore. You can get `dtheta_dsigma` from `material_utility`

Reply:
Same comments about `dsigma_dtheta` on this model

Reply:
Use it now